lib/Mojo/Promise.pm

```diff
@@ -214,8 +214,19 @@ sub _settle {
     $self = $self->new;
   }
 
+  return $self if $self->{semi_settled}++;
+
   if ($thenable && $status eq 'resolve') {
-    $results[0]->then(sub { $self->resolve(@_); () }, sub { $self->reject(@_); () });
+    if (refaddr $results[0] == refaddr $self) {
+      undef $self->{semi_settled};
+      $self->reject("Chaining cycle detected for promise $self");
```

Comment:
What is the use of including `$self` there?

Reply:
The use is that this kind of cycling error is only going to happen in a bug. So I included `$self` here to allow the developer to debug and figure out which of their promises took part in the cycle. (I'm assuming the developer can insert debugging code in their own code to display the memory address of their promises, but shouldn't have to tweak the Mojolicious library's code to display the address of the promise that took part in the cycle).

Reply:
Plus, node.js also displays `...for promise [Promise]` in their error message, which kind of reminds how objects are stringified (`[object Object]`), so perhaps including `$self` increases the consistency with node.js.